samtranslator/open_api/open_api.py

```diff
@@ -376,6 +379,75 @@ def add_tags(self, tags):
                 tag = {"name": name, self._X_APIGW_TAG_VALUE: value}
                 self.tags.append(tag)
 
+    def add_cors(
+        self,
+        allowed_origins,
+        allowed_headers=None,
+        allowed_methods=None,
+        expose_headers=None,
+        max_age=None,
+        allowed_credentials=None,
+    ):
+        """
+        Add CORS configuration to this Api to _X_APIGW_CORS header in open api definition
+
+        Following this guide:
+        https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-cors.html
+        https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigatewayv2-api-cors.html
+
+        :param list/dict allowed_origins: Comma separate list of allowed origins.
+            Value can also be an intrinsic function dict.
+        :param list/dict allowed_headers: Comma separated list of allowed headers.
+            Value can also be an intrinsic function dict.
+        :param list/dict allowed_methods: Comma separated list of allowed methods.
+            Value can also be an intrinsic function dict.
+        :param list/dict expose_headers: Comma separated list of allowed methods.
+            Value can also be an intrinsic function dict.
+        :param integer/dict max_age: Maximum duration to cache the CORS Preflight request. Value is set on
+            Access-Control-Max-Age header. Value can also be an intrinsic function dict.
+        :param bool/None allowed_credentials: Flags whether request is allowed to contain credentials.
+        """
+        ALLOW_ORIGINS = "allowOrigins"
+        ALLOW_HEADERS = "allowHeaders"
+        ALLOW_METHODS = "allowMethods"
+        EXPOSE_HEADERS = "exposeHeaders"
+        MAX_AGE = "maxAge"
+        ALLOW_CREDENTIALS = "allowCredentials"
+        cors_headers = [ALLOW_ORIGINS, ALLOW_HEADERS, ALLOW_METHODS, EXPOSE_HEADERS, MAX_AGE, ALLOW_CREDENTIALS]
+        cors_configuration = self._doc.get(self._X_APIGW_CORS, dict())
+
+        # intrinsics will not work if cors configuration is defined in open api and as a property to the HttpApi
+        if allowed_origins and is_intrinsic(allowed_origins):
+            cors_configuration_string = json.dumps(allowed_origins)
+            for header in cors_headers:
+                keyword = header[0].upper() + header[1:]
+                count = cors_configuration_string.count(keyword)
+                cors_configuration_string = cors_configuration_string.replace(keyword, header, count)
```

Comment:
Getting the count is redundant if you want to replace all occurrences of a string.

See https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/swagger/swagger.py#L285 for another place we did something similar.

Reply:
thank you, i didn't know replace will replace all occurrences. I will update it